### PR TITLE
EVG-13406 check for Not Found in manifest error message

### PR DESCRIPTION
**Pull request by ablack12**
In the previous fix for this, we assume that an APIRequestError has been returned.
https://github.com/evergreen-ci/evergreen/blob/7f3ad0d051fdb88d1b414d9c033c7866a5afcc31/service/api_plugin_manifest.go#L59
However, in the code itself, we hit this error, which is of the form APIResponseError, which has only a message.
https://github.com/evergreen-ci/evergreen/blob/7f3ad0d051fdb88d1b414d9c033c7866a5afcc31/thirdparty/github.go#L155
Really, the response is likely not nil, and if we correctly ran parseGithubErrorResponse(resp) then we would get a format that's more readable, and that the existing error check could understand. For simplicity, I moved this check into the existing initial `resp != nil` check for github functions.

### CODE REVIEW

File: thirdparty/github.go [hunk 1]
```diff
@@ -148,6 +148,9 @@ func GetGithubCommits(ctx context.Context, oauthToken, owner, repo, ref string,
 	commits, resp, err := client.Repositories.ListCommits(ctx, owner, repo, &options)
 	if resp != nil {
 		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
```
Comment (ybrill) [inline thirdparty/github.go:151]: I worry the status code could potentially be something else in the 200 series and not indicate an error (not a result of your changes).

Maybe an alternative is to parseGithubErrorResponse (assuming resp isn't nil) within the `if err != nil` block given `err != nil` implies the status code [isn't in the 200 series](https://github.com/google/go-github/blob/339ae9b3ec8be6eda90fa75a03bb1950b726d529/github/github.go#L785).

Comment (ybrill) [reply thirdparty/github.go:151]: I got this wrong. Only the inverse is true: if `err == nil` then the status code is in the 200 series.

Comment (ablack12) [reply thirdparty/github.go]: Good find! I've reorganized a bit

File: thirdparty/github.go [hunk 2]
```diff
@@ -196,23 +189,17 @@ func GetGithubFile(ctx context.Context, oauthToken, owner, repo, path, hash stri
...
-	}
-	if resp == nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return nil, FileNotFoundError{filepath: path}
+		}
+		if err != nil {
+			return nil, parseGithubErrorResponse(resp)
+		}
+	} else {
 		errMsg := fmt.Sprintf("nil response from github for '%s/%s' for '%s'", owner, repo, path)
```
Comment (ybrill) [inline thirdparty/github.go:199]: should this include err as well?

File: thirdparty/github.go [hunk 3]
```diff
@@ -337,17 +319,15 @@ func GetBranchEvent(ctx context.Context, oauthToken, repoOwner, repo, branch str
 	branchEvent, resp, err := client.Repositories.GetBranch(ctx, repoOwner, repo, branch)
 	if resp != nil {
 		defer resp.Body.Close()
-	}
-	if err != nil {
+		if err != nil {
+			return nil, parseGithubErrorResponse(resp)
+		}
+	} else {
 		errMsg := fmt.Sprintf("error querying  '%s/%s': branch: '%s': %v", repoOwner, repo, branch, err)
```
Comment (ybrill) [inline thirdparty/github.go:326]: should this say "nil response"?

**Review by ybrill — APPROVED** [review]

**Pull request merged**